packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -175,50 +185,28 @@ export default function MainFeedLayout({
     getFeatureValue(Features.FeedVersion, flags),
     10,
   );
-  const myFeedOnboardingVersion = getFeatureValue(
-    Features.MyFeedOnboardingVersion,
-    flags,
-  );
   const feedName = feedNameProp === 'default' ? defaultFeed : feedNameProp;
   const isMyFeed = feedName === MainFeedPage.MyFeed;
 
   const [isFirstSession, setIsFirstSession, isSessionLoaded] =
     usePersistentContext(FIRST_TIME_SESSION, isFirstVisit);
+  const [hasTriedOnboarding, setHasTriedOnboarding, hasOnboardingLoaded] =
+    usePersistentContext<boolean>(LOGGED_USER_ONBOARDING, false);
 
-  useEffect(() => {
-    if (user) {
-      setIsFirstSession(false);
-      setMyFeedMode(MyFeedMode.Manual);
-    } else if (
-      isFirstSession &&
-      isSessionLoaded &&
-      myFeedOnboardingVersion !== 'control'
-    ) {
-      setIsFirstSession(true);
-      setMyFeedMode(MyFeedMode.Auto);
-      setCreateMyFeed(true);
-    }
-  }, [isSessionLoaded, user]);
-
-  useEffect(() => {
-    if (
-      defaultFeed !== null &&
-      feedName !== null &&
-      feedName !== defaultFeed &&
-      !getShouldRedirect(isMyFeed, !!user)
-    ) {
-      updateDefaultFeed(feedName);
-    }
-  }, [defaultFeed, feedName]);
+  const modalCommand: ModalCommand = {
+    v1: setCreateMyFeed,
+    v2: setIsOnboardingOpen,
+  };
```

Comment:
I found it hard to understand at first what the ModalCommand exactly did.
Is there anyway we can make it a bit more clear?

Reply:
Will refactor this to the new hook/component.